depthcharge/data/hdf5.py

```diff
@@ -261,6 +262,52 @@ def get_spectrum(self, file_index, spectrum_index):
 
         return out
 
+    def get_spectrum_id(self, idx):
+        """Get the identifier for a mass spectrum.
+
+        Parameters
+        ----------
+        idx : int
+            The index of the mass spectrum in the SpectrumIndex.
+
+        Returns
+        -------
+        ms_data_file : str
+            The mass spectrometry data file from which the mass spectrum was
+            originally parsed.
+        identifier : str
+            The mass spectrum identifier, per PSI recommendations.
+        """
+        group_index, row_index = self._get_subindex(idx)
+        if self._handle is None:
+            raise RuntimeError("Use the context manager for access.")
+
+        grp = self._handle[str(group_index)]
+        ms_data_file = grp.attrs["path"]
+        identifier = grp["metadata"][row_index]["scan_id"].decode()
+        return ms_data_file, identifier
+
+    def _get_subindex(self, idx):
+        """Get the subindices to find a mass spectrum.
+
+        Parameters
+        ----------
+        idx : int
+            The overall index of the mass spectrum to retrieve.
+
+        Returns
+        -------
+        group_index : int
+            The group index of the HDF5 dataset in which to look.
+        row_index : int
+            The row within the dataset to find the mass spectrum.
+        """
+        group_index = np.searchsorted(
```

Comment:
**question:** What would be the performance implications of having to do a binary search to retrieve a batch of spectrum indexes (i.e. >1000 indexes in a loop), potentially for a large HDF5 file?

Reply:
You bring up a good point that it's probably pretty slow. I may open a new issue about this. I've been using [awkward-array](https://awkward-array.readthedocs.io/en/latest/) for some other projects recently; maybe I should dig around their code to see if they have a better solution.

I suppose the trivial solution would be to store a mapping of the 1D index to the group and row indices. Then we could get O(1) performance with the index finding step a least. For a reasonable number of spectra this would fit in-memory. 

I'll open a separate issue to explore this.

Reply:
I actually went ahead and implemented this mapping strategy. YOLO.